lib/std/os/linux/loongarch64.zig

```diff
@@ -0,0 +1,176 @@
+    );
+}
+
+pub fn syscall2(number: SYS, arg1: usize, arg2: usize) usize {
+    return asm volatile (
+        \\ syscall 0
+        : [ret] "={$r4}" (-> usize),
+        : [number] "{$r11}" (@intFromEnum(number)),
+          [arg1] "{$r4}" (arg1),
+          [arg2] "{$r5}" (arg2),
+        : "$t0", "$t1", "$t2", "$t3", "$t4", "$t5", "$t6", "$t7", "$t8", "memory"
+    );
+}
+
+pub fn syscall3(number: SYS, arg1: usize, arg2: usize, arg3: usize) usize {
+    return asm volatile (
+        \\ syscall 0
+        : [ret] "={$r4}" (-> usize),
+        : [number] "{$r11}" (@intFromEnum(number)),
+          [arg1] "{$r4}" (arg1),
+          [arg2] "{$r5}" (arg2),
+          [arg3] "{$r6}" (arg3),
+        : "$t0", "$t1", "$t2", "$t3", "$t4", "$t5", "$t6", "$t7", "$t8", "memory"
+    );
+}
+
+pub fn syscall4(number: SYS, arg1: usize, arg2: usize, arg3: usize, arg4: usize) usize {
+    return asm volatile (
+        \\ syscall 0
+        : [ret] "={$r4}" (-> usize),
+        : [number] "{$r11}" (@intFromEnum(number)),
+          [arg1] "{$r4}" (arg1),
+          [arg2] "{$r5}" (arg2),
+          [arg3] "{$r6}" (arg3),
+          [arg4] "{$r7}" (arg4),
+        : "$t0", "$t1", "$t2", "$t3", "$t4", "$t5", "$t6", "$t7", "$t8", "memory"
+    );
+}
+
+pub fn syscall5(number: SYS, arg1: usize, arg2: usize, arg3: usize, arg4: usize, arg5: usize) usize {
+    return asm volatile (
+        \\ syscall 0
+        : [ret] "={$r4}" (-> usize),
+        : [number] "{$r11}" (@intFromEnum(number)),
+          [arg1] "{$r4}" (arg1),
+          [arg2] "{$r5}" (arg2),
+          [arg3] "{$r6}" (arg3),
+          [arg4] "{$r7}" (arg4),
+          [arg5] "{$r8}" (arg5),
+        : "$t0", "$t1", "$t2", "$t3", "$t4", "$t5", "$t6", "$t7", "$t8", "memory"
+    );
+}
+
+pub fn syscall6(
+    number: SYS,
+    arg1: usize,
+    arg2: usize,
+    arg3: usize,
+    arg4: usize,
+    arg5: usize,
+    arg6: usize,
+) usize {
+    return asm volatile (
+        \\ syscall 0
+        : [ret] "={$r4}" (-> usize),
+        : [number] "{$r11}" (@intFromEnum(number)),
+          [arg1] "{$r4}" (arg1),
+          [arg2] "{$r5}" (arg2),
+          [arg3] "{$r6}" (arg3),
+          [arg4] "{$r7}" (arg4),
+          [arg5] "{$r8}" (arg5),
+          [arg6] "{$r9}" (arg6),
+        : "$t0", "$t1", "$t2", "$t3", "$t4", "$t5", "$t6", "$t7", "$t8", "memory"
+    );
+}
+
+const CloneFn = *const fn (arg: usize) callconv(.C) u8;
+
+pub extern fn clone(func: CloneFn, stack: usize, flags: u32, arg: usize, ptid: *i32, tls: usize, ctid: *i32) usize;
+
+pub const restore = restore_rt;
+
+pub fn restore_rt() callconv(.Naked) noreturn {
+    asm volatile (
+        \\ or $a7, $zero, %[number]
+        \\ syscall 0
+        :
+        : [number] "r" (@intFromEnum(SYS.rt_sigreturn)),
+        : "$t0", "$t1", "$t2", "$t3", "$t4", "$t5", "$t6", "$t7", "$t8", "memory"
+    );
+}
+
+pub const blksize_t = i32;
+pub const nlink_t = u32;
+pub const time_t = isize;
```

Comment:
For this group of types, best to write out the exact type they correspond to since we're in a bitness-specific file anyway. Just makes it a bit more obvious when you're skimming through the file.

Reply:
Fixed.